[GH-1322] Detect broken `yum install python-requests`

c-w opened the pull request:
## Detect issues with libcloud and python-requests on RHEL 7

### Description

As discussed in https://github.com/apache/libcloud/issues/1322, libcloud breaks when used on RHEL 7.6 with requests installed from yum instead of PyPI. In order to prevent user confusion, this pull request introduces a check for this environment state and provides guidance for the user on how to correctly set up their system for libcloud to function.

Resolves https://github.com/apache/libcloud/issues/1322

### Status

- done, ready for review

### Checklist (tick everything that applies)

- [x] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) ([build passed](https://travis-ci.org/CatalystCode/libcloud/builds/567156431))
- [x] Documentation (updated changelog)
- [x] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html) (added unit test)
- [x] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

c-w committed:
Detect broken `yum install python-requests`

c-w committed:
Update changelog

Kami commented:
Can you please also add a test case for it?

On a related note - we should also advise uses to use virtualenv with no system packages in such scenario (that's a good practice in general).

Kami commented on libcloud/__init__.py:
```diff
@@ -24,6 +24,8 @@
 import codecs
 import atexit
 
+import requests
```
We import this file in ``setup.py`` to read ``__version__`` attribute from it which means ``setup.py`` calls could fail if ``requests`` is not installed on the system (aka when setup.py is called during installations, it's unlikely these days, but still) - https://github.com/apache/libcloud/blob/trunk/setup.py#L76. It's generally a bad practice to do that, but that code has probably been there for a long time.

I think it would be safer and better for us to simply parse this file as string and use regex to match ``__version__`` attribute.

Something along those lines - https://github.com/StackStorm/st2/blob/master/scripts/dist_utils.py#L134

c-w replied:
In order to simplify the check and to stick with the existing pattern used for the paramiko library, I've changed the code to use a try/catch import. During `setup.py`, we don't particularly care about executing the requests version check (this can be delayed until first run of the library) so the functionality should be equivalent to what was there before. Let me know if this is okay.

c-w commented:
@Kami I believe I addressed all the comments. I also verified that `python setup.py install` works on a clean virtualenv. Could you please take another look and let me know if this is ready for merge? Thanks!

yairshemla committed:
Add support for "cpuPlatform" field in GCE driver

Kami committed:
Merge pull request #1343 from yairshemla/gce_driver_add_missing_cpu_platform_field

Add missing cpu platform field to GCE driver

Kami committed:
Add changelog entry for #1343.

Kami committed:
Add test case for change in #1343.

Kami committed:
Merge branch 'detect-broken-yum-python-requests' of https://github.com/CatalystCode/libcloud into CatalystCode-detect-broken-yum-python-requests

Kami committed:
Also assert on the exception error message, add additional test case.

Kami commented:
Sorry for the delay. I pushed a small change (8d179ed9d5402988e4e5138aa3db51befc634274), besides that it looks good to me :+1:

Kami commented:
@c-w a lot of changes have accumulated in trunk since the last release.

Do you want to start the work needed to get the 2.6.0 release out - https://libcloud.readthedocs.io/en/latest/committer_guide.html#making-a-release-for-release-managers?
